gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -391,18 +391,30 @@ private static void setupConjureUndertowProject(
         }
     }
 
+    @SuppressWarnings("unchecked")
     private static void ignoreFromCheckUnusedDependencies(Project proj) {
-        proj.getPluginManager().withPlugin("com.palantir.baseline-exact-dependencies", plugin -> {
-            Object task = proj.getTasks().getByName("checkUnusedDependencies");
+        proj.getPlugins().withId("com.palantir.baseline-exact-dependencies", plugin -> {
+            Class<? extends Task> checkUnusedDependenciesTask;
             try {
-                Method ignoreMethod = task.getClass().getMethod("ignore", String.class, String.class);
-                List<String> conjureJavaLibComponents = Splitter.on(':').splitToList(CONJURE_JAVA_LIB_DEP);
-                ignoreMethod.invoke(task, conjureJavaLibComponents.get(0), conjureJavaLibComponents.get(1));
-                // also ignore guava since retrofit adds it...
-                ignoreMethod.invoke(task, "com.google.guava", "guava");
-            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                ClassLoader baselineClassloader = plugin.getClass().getClassLoader();
```

Comment:
this is a neat way to get the right classloader, whether baseline is applied via `plugins{}` buildscript.

Reply:
Why not just disable the entire task on conjure subprojects?

Reply:
Have considered this too, but ultimately I prefer to still restrict the jars people depend on, especially for api modules, which are supposed to be quite light.